map_test.go

```diff
@@ -378,6 +378,27 @@ func TestNestedMapPin(t *testing.T) {
 	defer m.Close()
 }
 
+func TestNestedMapPinNested(t *testing.T) {
+	expError := errors.New("inner maps cannot be pinned")
+
+	if _, err := NewMap(&MapSpec{
+		Type:       ArrayOfMaps,
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: 2,
+		InnerMap: &MapSpec{
+			Name:       "inner",
+			Type:       Array,
+			KeySize:    4,
+			ValueSize:  4,
+			MaxEntries: 1,
+			Pinning:    PinByName,
+		},
+	}); err == nil || err.Error() != expError.Error() {
```

Comment:
I think the below is enough:

```suggestion
	}); err == nil {
```

If you prefer to be more thorough I'd recommend introducing a global but unexported `var errPinnedInnerMap = ...` and then check that here.

Reply:
done